gnovm/pkg/doctest/exec.go

```diff
@@ -0,0 +1,179 @@
+package doctest
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"path/filepath"
+	"regexp"
+	"runtime"
+	"strings"
+	"sync"
+
+	"github.com/gnolang/gno/gno.land/pkg/sdk/vm"
+	bft "github.com/gnolang/gno/tm2/pkg/bft/types"
+	"github.com/gnolang/gno/tm2/pkg/crypto"
+	"github.com/gnolang/gno/tm2/pkg/db/memdb"
+	"github.com/gnolang/gno/tm2/pkg/log"
+	"github.com/gnolang/gno/tm2/pkg/sdk"
+	authm "github.com/gnolang/gno/tm2/pkg/sdk/auth"
+	bankm "github.com/gnolang/gno/tm2/pkg/sdk/bank"
+	"github.com/gnolang/gno/tm2/pkg/std"
+	"github.com/gnolang/gno/tm2/pkg/store"
+	"github.com/gnolang/gno/tm2/pkg/store/dbadapter"
+	"github.com/gnolang/gno/tm2/pkg/store/iavl"
+)
+
+// Option constants
+const (
+	IGNORE       = "ignore"       // Do not run the code block
+	SHOULD_PANIC = "should_panic" // Expect a panic
+	ASSERT       = "assert"       // Assert the result and expected output are equal
+)
+
+func GetStdlibsDir() string {
+	_, filename, _, ok := runtime.Caller(0)
+	if !ok {
+		panic("cannot get current file path")
+	}
+	return filepath.Join(filepath.Dir(filename), "..", "..", "stdlibs")
+}
+
+// cache stores the results of code execution.
+var cache struct {
+	m map[string]string
+	sync.RWMutex
+}
+
+func init() {
+	cache.m = make(map[string]string)
+}
+
+// hashCodeBlock generates a SHA256 hash for the given code block.
+func hashCodeBlock(c codeBlock) string {
+	h := sha256.New()
+	h.Write([]byte(c.content))
+	return hex.EncodeToString(h.Sum(nil))
+}
+
+// ExecuteCodeBlock executes a parsed code block and executes it in a gno VM.
+func ExecuteCodeBlock(c codeBlock, stdlibDir string) (string, error) {
+	if c.lang == "go" {
+		c.lang = "gno"
```

Comment:
since we don't have gno fully recognized by github atm, i suggest we start having something like

````
```go,gnodoctest
package foo

func main() {
	a := 42
}
```
````

Reply:
It sounds good. I think we could consider setting execution options like `ignore` or `should_panic` for that feature.

Reply:
Changing 'go' to 'gno' (or using `gno`) seems fine as it's intended to specify the file extension for temporary executable files. 

However, I'm slightly concerned that including multiple elements on a single line when specifying the tag might break syntax highlighting during document rendering, especially using `go` language tag. 

If this isn't an issue, it might be useful to support specifying execution options like `ignore` or `should_panic`.